specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+
+For more information about the relation between a profile specification and the
+quantum instruction set we refer to the paragraph on [Bullet 1](#base-profile)
+in the introduction of this document. For more information about how and when
+the QIS is resolved, as well as recommendations for front- and backend
+developers, we refer to the document on [compilation stages and
+targeting](../Compilation_And_Targeting.md).
+
+## Output Recording
+
+The program output of a quantum application is defined by a sequence of calls to
+runtime functions that return values produced by the computation. In the case of
+the Base Profile, these calls are contained within the last block of the entry
+point function. LLVM provides a convenient set of tools to walk and visualize
+the control flow graph of a program. The last block(s) in a function are easy to
+find by looking for blocks without a
+[successor](https://releases.llvm.org/13.0.1/docs/ProgrammersManual.html#iterating-over-predecessors-successors-of-blocks).
+
+The following functions are can be used to record the program output:
+
+| Function                            | Signature             | Description                                                                                                                                                                                                                                |
+| :---------------------------------- | :-------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| __quantum__rt__tuple_record_output  | `void(i64,i8*)`      | Inserts a marker in the output log that indicates the start of a tuple and how many tuple elements are logged. The second parameter defines a string label for the tuple. Depending on the output schema, the label is logged or omitted.  |
+| __quantum__rt__array_record_output  | `void(i64,i8*)`      | Inserts a marker in the output log that indicates the start of an array and how many array elements are logged. The second parameter defines a string label for the array. Depending on the output schema, the label is logged or omitted. |
+| __quantum__rt__result_record_output | `void(%Result*,i8*)` | Adds a measurement result to the output log. The second parameter defines a string label for the result value. Depending on the output schema, the label is logged or omitted.                                                             |
+
+For all output recording functions, the `i8*` argument must be a non-null
+pointer to a global constant that contains a null-terminated string. A backend
+may ignore that argument if it guarantees that the order of the recorded output
+matches the order defined by the entry point. Conversely, certain output schemas
+do not require that the recorded output is listed in a particular order. For
```

Comment:
"do not require that the recorded output is" => "do not require the recorded output to be"?

Reply:
agreed, sounds better. done